source/fundamentals/gridfs.txt

```diff
@@ -0,0 +1,707 @@
+suitable for your use case, see :manual:`GridFS </core/gridfs>` in the
+{+mdb-server+} manual.
+
+How GridFS Works
+----------------
+
+GridFS organizes files in a **bucket**, a group of MongoDB collections
+that contain the chunks of files and information describing them. The
+bucket contains the following collections, named using the convention
+defined in the GridFS specification:
+
+- ``chunks``: Stores the binary file chunks
+- ``files``: Stores the file metadata
+
+The driver creates the GridFS bucket, if it doesn't exist, when you first
+write data to it. The bucket contains the preceding collections
+prefixed with the default bucket name ``fs``, unless you specify a different
+name. To ensure efficient retrieval of the files and related metadata, the driver
+creates an index on each collection. The driver ensures that these indexes exist
+before performing read and write operations on the GridFS bucket.
+	
+For more information about GridFS indexes, see :manual:`GridFS Indexes
+</core/gridfs/#gridfs-indexes>` in the {+mdb-server+} manual.
+
+When using GridFS to store files, the driver splits the files into smaller
+chunks, each represented by a separate document in the ``chunks`` collection.
+It also creates a document in the ``files`` collection that contains
+a file ID, file name, and other file metadata. You can upload the file by passing
+a stream to the {+driver-short+} to consume or by creating a new stream and writing to it
+directly.
+
+The following diagram shows how GridFS splits files when they are
+uploaded to a bucket:
+
+.. figure:: /includes/figures/GridFS-upload.png
+   :alt: A diagram that shows how GridFS uploads a file to a bucket
+
+When retrieving files, GridFS fetches the metadata from the ``files``
+collection in the specified bucket and uses the information to reconstruct
+the file from documents in the ``chunks`` collection. You can read the file
+by writing its contents to an existing stream or by creating a new stream that points
+to the file.
+
+.. _gridfs-create-bucket:
+
+Create a GridFS Bucket
+----------------------
+
+To begin using GridFS to store or retrieve files, create a new instance of the
+``GridFSBucket`` class, passing in an ``IMongoDatabase`` object that represents your
+database. This method accesses an existing bucket or creates
+a new bucket if one does not exist.
+
+The following example creates a new instance of the ``GridFSBucket`` class for the
+``db`` database:
+
+.. literalinclude:: /includes/code-examples/GridFS.cs
+   :language: csharp
+   :dedent:
+   :start-after: // start-create-bucket
+   :end-before: // end-create-bucket
+  
+.. _gridfs-create-custom-bucket:
+
+Customize the Bucket
+~~~~~~~~~~~~~~~~~~~~
+
+You can customize the GridFS bucket configuration by passing an instance
+of the ``GridFSBucketOptions class`` class to 
+the ``GridFSBucket()`` constructor. The following table describes the properties in the
+``GridFSBucketOptions`` class:
```

Comment:
```suggestion
the ``GridFSBucket()`` constructor. The following table describes the settable properties in the
``GridFSBucketOptions`` class:
```

Reply:
This might be a language-specific thing, but in my experience C# properties are assumed to be both gettable and settable unless otherwise specified as read-only (or, rarely, write-only)